src/main/java/net/minecraftforge/fml/common/asm/transformers/EventSubscriberTransformer.java

```diff
@@ -0,0 +1,73 @@
+package net.minecraftforge.fml.common.asm.transformers;
+
+import java.lang.reflect.Modifier;
+import java.util.List;
+
+import net.minecraft.launchwrapper.IClassTransformer;
+
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.AnnotationNode;
+import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.MethodNode;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+
+public class EventSubscriberTransformer implements IClassTransformer
+{
+    @Override
+    public byte[] transform(String name, String transformedName, byte[] basicClass)
+    {
+        if (basicClass == null) return null;
+
+        ClassNode classNode = new ClassNode();
+        new ClassReader(basicClass).accept(classNode, 0);
+
+        boolean isSubscriber = false;
+        List<MethodNode> nowOverridable = Lists.newArrayList();
+
+        for (MethodNode methodNode : classNode.methods)
+        {
+            List<AnnotationNode> anns = methodNode.visibleAnnotations;
+
+            if (anns != null && Iterables.any(anns, SubscribeEventPredicate.INSTANCE))
+            {
+                isSubscriber = true;
+                if (Modifier.isPrivate(methodNode.access)) nowOverridable.add(methodNode);
+                methodNode.access = toPublic(methodNode.access);
+            }
+        }
+
+        if (isSubscriber)
+        {
+            classNode.access = toPublic(classNode.access);
+            AccessTransformer.replaceInvokeSpecial(classNode, nowOverridable);
+
+            ClassWriter writer = new ClassWriter(ClassWriter.COMPUTE_MAXS);
+            classNode.accept(writer);
+            return writer.toByteArray();
+        }
+
+        return basicClass;
+    }
+
+    private static int toPublic(int access)
+    {
+        return access & ~(Opcodes.ACC_PRIVATE + Opcodes.ACC_PROTECTED) | Opcodes.ACC_PUBLIC;
+    }
+
+    private enum SubscribeEventPredicate implements Predicate<AnnotationNode>
+    {
+        INSTANCE;
```

Comment:
I REALLY hate this style of singeltons, yes its nice, but enums come with a lot of things attached. Just create a class and instance.